### PR TITLE
created reactnative search-bar example

### PR DESCRIPTION
**Pull request by cmreddy071**
- Created reactnative search-bar example. 
- Written test cases.

### CODE REVIEW

**Comment by cmreddy071** [comment]
- Use the searchableConfig from the header component.
- Fix the alignment issues.

File: App.js [hunk 1]
```diff
@@ -1,19 +1,98 @@
...
-    </View>
-  );
+import { StyleSheet, Text, View, FlatList } from 'react-native';
+import * as Colors from '@pxblue/colors'
+import { Header, wrapIcon } from '@pxblue/react-native-components';
+import MatIcon from 'react-native-vector-icons/MaterialIcons';
+import { ListItem } from 'react-native-elements';
+
+import sampleData from './data';
+import AnimatedSearchBar from './AnimatedSearchBar';
```
Comment (joebochill) [inline App.js]: remove this reference

File: App.js [hunk 2]
```diff
@@ -1,19 +1,98 @@
...
+
+const MenuIcon = wrapIcon({ IconClass: MatIcon, name: 'menu' });
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { showSearch: false, query: '', results: sampleData, data: sampleData };
```
Comment (joebochill) [inline App.js]: don't need this logic for showsearch - it's built into the header component

File: App.js [hunk 3]
```diff
@@ -1,19 +1,98 @@
...
+  }
+
+  onSearchToggle = () => this.setState((prevState) => ({ showSearch: !prevState.showSearch }));
```
Comment (joebochill) [inline App.js]: remove this function

File: App.js [hunk 4]
```diff
@@ -1,19 +1,98 @@
...
+        item.took_office.toLowerCase().trim().indexOf(query) >= 0
+      ) {
+        results.push(item);
+      }
+    });
+    this.setState({ results });
+  }
+  onClear = () => {
+    this.setState({ results: this.state.data, query: '', showSearch: false });
+  }
```
Comment (joebochill) [inline App.js]: remove this function

File: App.js [hunk 5]
```diff
@@ -1,19 +1,98 @@
...
+            placeholder: 'Search',
+            onChangeText: (q) => this.onSearchChange(q)
+          }}
+        />
+        <AnimatedSearchBar
+          show={this.state.showSearch}
+          value={this.state.query}
+          onClear={this.onClear}
+          onChangeText={this.onSearchChange}
+        />
```
Comment (joebochill) [inline App.js]: remove this component

File: App.js [hunk 6]
```diff
@@ -1,19 +1,98 @@
...
+                data={this.state.results}
+                keyExtractor={(item, index) => `${index}`}
+                renderItem={({ item }) => (
+                  <ListItem
+                    title={item.president}
+                    subtitle={(<View style={{ marginLeft: 15 }}>
+                      <Text style={{ color: Colors.gray[500] }}>{item.party}</Text>
+                      <Text style={{ color: Colors.gray[500] }}>{item.took_office}</Text>
+                    </View>)}
+                    titleStyle={{ marginLeft: 15 }}
```
Comment (joebochill) [inline App.js]: these left margins should be 16 not 15

File: App.js [hunk 7]
```diff
@@ -1,19 +1,98 @@
...
+                    leftIcon={{ name: 'person', color: Colors.gray[500], iconStyle: { marginLeft: 3 } }}
```
Comment (joebochill) [inline App.js]: why marginLeft 3?

File: App.js [hunk 8]
```diff
@@ -1,19 +1,98 @@
...
+  onClear = () => {
+    this.setState({ results: this.state.data, query: '', showSearch: false });
+  }
+  render() {
+    return (
+      <View style={styles.container}>
+        <Header
+          title={'President'}
+          subtitle={'Leader of the Free world'}
+          backgroundColor={Colors.blue[500]}
```
Comment (joebochill) [inline App.js]: don't manually set the color...use the theme

File: AnimatedSearchBar.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { StyleSheet, Animated, Dimensions, Easing } from 'react-native';
+import * as Colors from '@pxblue/colors'
+import { SearchBar, Icon } from 'react-native-elements';
+import { getStatusBarHeight } from 'react-native-status-bar-height';
+
+const { width: SCREEN_WIDTH } = Dimensions.get('window');
+
+class AnimatedSearchBar extends React.PureComponent {
```
[the remark this answers is not in the document]
Comment (cmreddy071) [reply AnimatedSearchBar.js]: File deleted

File: App.js (hunk 1 above)
Comment (cmreddy071) [reply App.js]: Removed

File: App.js (hunk 2 above)
Comment (cmreddy071) [reply App.js]: Removed the logic for show search

File: App.js (hunk 3 above)
Comment (cmreddy071) [reply App.js]: Function removed.

File: App.js (hunk 4 above)
Comment (cmreddy071) [reply App.js]: Function removed.

File: App.js (hunk 5 above)
Comment (cmreddy071) [reply App.js]: Component removed.

File: App.js (hunk 6 above)
Comment (cmreddy071) [reply App.js]: marginLeft changed to 16px

File: App.js (hunk 8 above)
Comment (cmreddy071) [reply App.js]: Used PXB Theme.

**Comment by cmreddy071** [comment]
Added Focus to searchbar.

File: tests/App.test.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+Enzyme.configure({ adapter: new Adapter() })
+
+describe('App Tests ', function () {
+    it('App Renders', () => {
+        const tree = renderer.create(
+            <App />
+        ).toJSON();
+        console.log("tree", tree)
+        expect(tree).toMatchSnapshot();
+    });
```
Comment (joebochill) [inline tests/App.test.js]: if you're going to snapshot test, you should also add one for when the search header is open

Comment (joebochill) [reply tests/App.test.js]: I take it back, those tests are accounted for in the component so no need to duplicate them

**Comment by cmreddy071** [comment]
Applied font family to the list elements but semiBold is not working.

**Comment by cmreddy071** [comment]
Updated themes version to 2.2.4.

File: tests/App.test.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+Enzyme.configure({ adapter: new Adapter() })
+
+describe('App Tests ', function () {
+    it('App Renders', () => {
+        const tree = renderer.create(
+            <App />
+        ).toJSON();
+        console.log("tree", tree)
+        expect(tree).toMatchSnapshot();
```
Comment (joebochill) [inline tests/App.test.js]: this snapshot test is broken....the tree is ```null```

File: tests/__snapshots__/App.test.js.snap [hunk 12]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`App Tests  App Renders 1`] = `null`;
```
Comment (joebochill) [inline tests/__snapshots__/App.test.js.snap]: This is not what the snapshot JSON tree should look like

**Pull request merged**